### PR TITLE
Consolidate CMake behavior for SimpleITK_MAX_DIMENSION configuration

### PR DESCRIPTION
**Pull request by blowekamp**

### CODE REVIEW

File: CMake/sitkMaxDimensionOption.cmake [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  if (NOT SimpleITK_4D_Image)
+    set( SimpleITK_MAX_DIMENSION_DEFAULT 3 )
+  endif()
+endif()
+
+set( "SimpleITK_MAX_DIMENSION" "${SimpleITK_MAX_DIMENSION_DEFAULT}"
+  CACHE STRING "Add Image and I/O support up to max dimension (integer)." )
+mark_as_advanced( SimpleITK_MAX_DIMENSION )
+
+if (NOT SimpleITK_MAX_DIMENSION MATCHES "^[0-9]+$")
```
Comment (zivy) [inline CMake/sitkMaxDimensionOption.cmake:25]: I think the regular expression should be "^[1-9][0-9]+$" number shouldn't start with a zero.

**Review by zivy — COMMENTED** [review]
Minor issue with regular expression.

File: CMake/sitkMaxDimensionOption.cmake (hunk 1 above)
Comment (blowekamp) [reply CMake/sitkMaxDimensionOption.cmake:25]: Accepting leading zeros for numbers is standard numerical input behavior. The check for a minimum value is done below. I don't see any potential problem.

**Review by zivy — APPROVED** [review]

**Review by dave3d — APPROVED** [review]
LGTM

**Pull request merged**